python/lsst/meas/algorithms/detection.py

```diff
@@ -366,11 +370,36 @@ def detectFootprints(self, exposure, doSmooth=True, sigma=None, clearMask=True):
 
         fpSets = pipeBase.Struct(positive=None, negative=None)
 
+        if self.config.doTempLocalBackground:
+            # Subtract the local background from the smoothed image.
```

Comment:
One could imagine using a pretty small-scale background, so I'm not sure your assumption holds.